PlaceRouteHierFlow/placer/Placer.cpp

```diff
@@ -241,6 +242,28 @@ std::map<double, std::pair<SeqPair, ILP_solver>> Placer::PlacementCoreAspectRati
     logger->debug("Random number generator seed={0}", seed);
   }
 
+  if (curr_sp.Enumerate()) {
+    const auto maxcount = size_t(1. * log(hyper.T_MIN / hyper.T_INT) / log(hyper.ALPHA));
+    size_t cnt{0};
+    while (!curr_sp.EnumExhausted()) {
+      ILP_solver tsol(designData, hyper.ilp_solver);
+      curr_cost = tsol.GenerateValidSolution(designData, curr_sp, drcInfo, hyper.NUM_THREADS);
+      tsol.cost = curr_cost;
+      logger->debug("Solving enumeration {0} cost {1}", cnt, curr_cost);
+      if (curr_cost >= 0) {
+        oData[curr_cost] = std::make_pair(curr_sp, tsol);
+        ReshapeSeqPairMap(oData, nodeSize);
+      }
+      curr_sp.PerturbationNew(designData);
+      ++cnt;
+      if (cnt >= maxcount) break;
```

Comment:
1. After `maxcount` trials, do we want to fall back to Simulated Annealing for another `maxcount` trials? Should we just throw an assertion instead? 
2. If we still want to resort to Simulated Annealing, should we cache these enumerations so that we do not solve ILP again?

Reply:
The cost comparisons in this method do not seem consistent, both >=0 and >0 are used.

Reply:
I think `assert()` makes sense here. Since there aren't any more sequence pairs left to explore.